CONTRIBUTING.md

```diff
@@ -110,3 +105,68 @@ You need to be in the directory in which the images are defined and have a `resi
 find . -name '*-gray.svg' -type f -exec inkscape {} -w 72 -h 72 --export-filename resize/{}.png \; && find ./resize/ -type f -name '*.svg.png' -exec sh -c 'f="{}"; mv -- "$f" "${f%.svg.png}.png"' \;
 ```
 
+### Code Architecture
+
+In this section we'll explain the main packages if you want to contribute/read/understand how InfraMap does what it does.
+
+#### Providers
+
+The main idea behind InfraMap is to abstract each specific provider logic into a different implementation with a common interface
```

Comment:
This para is a bit redundant with himself. I would reword in in this way:

> The main idea behind InfraMap is to abstract each specific Terraform provider logic into a different implementation using a common interface: `provider.Provider`.
All this implementations are inside the `provider/` package.